src/test/java/it/pagopa/selfcare/pagopa/backoffice/service/ApiManagementServiceTest.java

```diff
@@ -0,0 +1,173 @@
+package it.pagopa.selfcare.pagopa.backoffice.service;
+
+import it.pagopa.selfcare.pagopa.backoffice.TestUtil;
+import it.pagopa.selfcare.pagopa.backoffice.client.AzureApiManagerClient;
+import it.pagopa.selfcare.pagopa.backoffice.client.ExternalApiClient;
+import it.pagopa.selfcare.pagopa.backoffice.model.institutions.*;
+import it.pagopa.selfcare.pagopa.backoffice.model.institutions.client.InstitutionApiKeys;
+import it.pagopa.selfcare.pagopa.backoffice.model.institutions.client.InstitutionInfo;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@WithMockUser(username = "user1", password = "pwd", roles = "USER")
+class ApiManagementServiceTest {
+
+    @MockBean
+    private AzureApiManagerClient apimClient;
+
+    @MockBean
+    private ExternalApiClient externalApiClient;
+
+    @SpyBean
+    private ModelMapper modelMapper;
+
+    @Autowired
+    @InjectMocks
+    private ApiManagementService service;
+
+    private final String INSTITUTION_ID = "INSTITUTION_ID";
+
+    private final String SUBSCRIPTION_ID = "SUBSCRIPTION_ID";
+
+    private final String BROKER_ID = "BROKER_ID";
+
+    @Test
+    void getInstitutions() {
+        when(externalApiClient.getInstitutions(any()))
+                .thenReturn(Collections.singletonList(new InstitutionInfo()));
+        List<InstitutionDetail> institutions = service.getInstitutions();
+        assertNotNull(institutions);
+        assertFalse(institutions.isEmpty());
+        verify(externalApiClient).getInstitutions(any());
+    }
+
+    @Test
+    void getInstitution() throws IOException {
+        when(externalApiClient.getInstitution(any()))
+                .thenReturn(TestUtil.fileToObject(
+                        "response/externalapi/institution_response.json", InstitutionResponse.class));
+        Institution institution = service.getInstitution(INSTITUTION_ID);
+        assertNotNull(institution);
+        verify(externalApiClient).getInstitution(any());
```

Comment:
some advice: you can consider to use the [captor](https://github.com/pagopa/pagopa-selfcare-ms-backoffice-backend/blob/2bbe63ec87f497681103a8b5fe925073ff82dce2/src/test/java/it/pagopa/selfcare/pagopa/backoffice/scheduler/CiBrokerExtractionSchedulerTest.java) to do a deep check